src/main/webapp/resources/js/apis/samples/samples.js

```diff
@@ -31,14 +34,77 @@ export const sampleApi = createApi({
       }),
       invalidatesTags: ["SampleDetails"],
     }),
+    addSampleMetadata: build.mutation({
+      query: ({
+        sampleId,
+        projectId,
+        metadataField,
+        metadataEntry,
+        metadataRestriction,
+      }) => ({
+        url: `/${sampleId}/metadata`,
+        body: { projectId, metadataField, metadataEntry, metadataRestriction },
+        method: "POST",
+      }),
+      invalidatesTags: ["SampleMetadata"],
+    }),
+    removeSampleMetadata: build.mutation({
+      query: ({ field, entryId, projectId }) => ({
+        url: `/metadata?projectId=${projectId}&metadataField=${field}&metadataEntryId=${entryId}`,
+        method: "DELETE",
+      }),
+      invalidatesTags: ["SampleMetadata"],
+    }),
+    updateSampleMetadata: build.mutation({
+      query: ({
+        sampleId,
+        projectId,
+        metadataFieldId,
+        metadataField,
+        metadataEntryId,
+        metadataEntry,
+        metadataRestriction,
+      }) => ({
+        url: `/${sampleId}/metadata`,
+        body: {
+          projectId,
+          metadataFieldId,
+          metadataField,
+          metadataEntryId,
+          metadataEntry,
+          metadataRestriction,
+        },
+        method: "PUT",
+      }),
+      invalidatesTags: ["SampleMetadata"],
+    }),
   }),
 });
 
 export const {
   useGetSampleDetailsQuery,
   useUpdateSampleDetailsMutation,
+  useAddSampleMetadataMutation,
+  useRemoveSampleMetadataMutation,
+  useUpdateSampleMetadataMutation,
 } = sampleApi;
 
+/**
+ * Gets the sample metadata.
+ * @param {Object} params
+ * @returns {Promise<{}|T>}
+ */
+export const fetchMetadataForSample = async ({ sampleId, projectId }) => {
```

Comment:
Shouldn't this be part of the API for a sample?  With a mutation that triggers the update?

Reply:
As discussed in the SampleMetadata.jsx file we are using a selector to access the metadata from the redux store and using a useeffect hook to update the state. Having this as a part of the sample API resulted in the metadata always returning an empty array, so just followed how we have handled stuff like this in the past